packages/inference/src/providers/wavespeed-ai.ts

```diff
@@ -0,0 +1,193 @@
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import { ImageToImageArgs } from "../tasks";
+import type { BodyParams, HeaderParams, RequestArgs, UrlParams } from "../types";
+import { delay } from "../utils/delay";
+import { omit } from "../utils/omit";
+import { base64FromBytes } from "../utils/base64FromBytes";
+import {
+	TaskProviderHelper,
+	TextToImageTaskHelper,
+	TextToVideoTaskHelper,
+	ImageToImageTaskHelper,
+} from "./providerHelper";
+
+const WAVESPEEDAI_API_BASE_URL = "https://api.wavespeed.ai";
+
+/**
+ * Common response structure for all WaveSpeed AI API responses
+ */
+interface WaveSpeedAICommonResponse<T> {
+	code: number;
+	message: string;
+	data: T;
+}
+
+/**
+ * Response structure for task status and results
+ */
+interface WaveSpeedAITaskResponse {
+	id: string;
+	model: string;
+	outputs: string[];
+	urls: {
+		get: string;
+	};
+	has_nsfw_contents: boolean[];
+	status: "created" | "processing" | "completed" | "failed";
+	created_at: string;
+	error: string;
+	executionTime: number;
+	timings: {
+		inference: number;
+	};
+}
+
+/**
+ * Response structure for initial task submission
+ */
+interface WaveSpeedAISubmitResponse {
+	id: string;
+	urls: {
+		get: string;
+	};
+}
+
+type WaveSpeedAIResponse<T = WaveSpeedAITaskResponse> = WaveSpeedAICommonResponse<T>;
```

Comment:
I'm not sure this type alias is needed, can we remove it?
```suggestion
```

`WaveSpeedAICommonResponse` can be renamed to `WaveSpeedAIResponse`

Reply:
This type is needed and will be used in two places. It's uncertain whether it will be used again in the future.
It follows the DRY (Don't Repeat Yourself) principle
It provides better type safety (through default generic parameters)
It makes the code more readable and maintainable

Reply:
This type is needed and will be used in two places. It's uncertain whether it will be used again in the future.
It follows the DRY (Don't Repeat Yourself) principle
It provides better type safety (through default generic parameters)
It makes the code more readable and maintainable